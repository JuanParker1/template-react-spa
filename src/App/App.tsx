import { HashRouter } from "react-router-dom";

import { GlobalProvider } from "~/contexts";
import { Routes } from "~/routes/Routes";
import { theme } from "~/styles/mui/theme";
import { useRegisterSW } from "virtual:pwa-register/react";

import {
  CssBaseline,
  StyledEngineProvider,
  ThemeProvider,
} from "@mui/material";

function App() {
  useRegisterSW();

  return (
    <HashRouter>
      <GlobalProvider>
        <ThemeProvider theme={theme}>
          <StyledEngineProvider injectFirst>
            <CssBaseline />
            <Routes />
          </StyledEngineProvider>
        </ThemeProvider>
      </GlobalProvider>
    </HashRouter>
  );
}

export { App };
