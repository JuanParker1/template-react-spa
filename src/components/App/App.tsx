import { HashRouter } from "react-router-dom";

import { Routes } from "~/routes/Routes";
import { theme } from "~/styles/theme";
import { useRegisterSW } from "virtual:pwa-register/react";

import { CssBaseline, ThemeProvider } from "@mui/material";

function App() {
  useRegisterSW();

  return (
    <HashRouter>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        <Routes />
      </ThemeProvider>
    </HashRouter>
  );
}

export { App };
