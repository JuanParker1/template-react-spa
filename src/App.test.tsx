import App from "~/App";
import { shallow } from "enzyme";
import { describe, expect, it } from "vitest";

describe("App", () => {
  it("renders snapshots", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
