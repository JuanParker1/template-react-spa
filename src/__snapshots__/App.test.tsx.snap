// Vitest Snapshot v1

exports[`App > renders snapshots 1`] = `ShallowWrapper {}`;
